custom_components/birthdays/__init__.py

```diff
@@ -45,28 +52,41 @@ async def async_setup(hass, config):
     tasks = [asyncio.create_task(device.update_data()) for device in devices]
     await asyncio.wait(tasks)
 
+    _LOGGER.debug(devices)
+
     return True
 
 
 class BirthdayEntity(Entity):
 
-    def __init__(self, name, date_of_birth, icon, hass):
+    def __init__(self, unique_id, name, date_of_birth, icon, attributes, hass):
         self._name = name
-        self._date_of_birth = date_of_birth
-        self._icon = icon
-        self._age_at_next_birthday = 0
-        self._state = None
-        name_in_entity_id = slugify(name)
-        self.entity_id = 'birthday.{}'.format(name_in_entity_id)
-        self.hass = hass
+
+        if unique_id is not None:
+            self._unique_id = slugify(unique_id)
+        else: 
+            self._unique_id = slugify(name)
+
+        self._state                = None
+        self._icon                 = icon
```

Comment:
Why the many spaces in between assignments here? All of these self-variables could look like `self._icon = icon` and still be readable :)

Reply:
Good point, removed! :-)